Source/WebCore/platform/graphics/SourceBufferPrivate.cpp

```diff
@@ -260,10 +275,12 @@ void SourceBufferPrivate::clearTrackBuffers(bool shouldReportToClient)
         return;
 
     updateHighestPresentationTimestamp();
-    updateBufferedFromTrackBuffers(true);
 
-    if (isAttached())
+    if (isAttached()) {
+        m_client->sourceBufferPrivateTrackBuffersChanged({ });
         m_client->sourceBufferPrivateReportExtraMemoryCost(totalTrackBufferSizeInBytes());
+    }
+    updateBufferedFromTrackBuffers({ }, true);
```

Comment:
Preexisting, but it might be good to use a temp variable isEnded = true to improve readability.